### DISCUSSION TITLE
Install in a separate project and call into main site via sub-domain?

### DISCUSSION THREAD

**Discussion by jerryhall** [austintoddj/canvas#949] [General]
The tldr; is in the subject line. 

My question is, (being new to Laravel but, not web design) I'd love to include a blog in a website I'm building for a separate Laravel app project. 

Would it make sense to create a separate project exclusively focused on Canvas and accessed through a sub-domain, away from my main app project? 

My only, perhaps unfounded, fear is my commingling the two projects files to such a point that it wasn't clear which controller or page was for which project.

Thanks for any feedback!

**Comment by austintoddj** [comment]
Thanks for checking out Canvas!

My advice would be to keep things as simple as possible, and use one Laravel app. Since Canvas itself will reside in the `vendor` directory, the only 2 files that'll actually be published to your project will be:

- `config/canvas.php`
- `app/Providers/CanvasServiceProvider.php`

Now, if you choose to use the `canvas-ui` Artisan command to generate a frontend for you, then you'll be provided with some views, controllers, JS and more. All of which are properly namespaced, so you shouldn't have too much confusion as to what everything is.

Going back to the beginning, you can simply drop your subdomain in the config file. I'd also recommend then setting the `CANVAS_PATH ` variable in your .env file to `null`, just because the path defaults to `/canvas`.

Biggest wins here are that you've only got one deploy to worry about now, and one app to keep upgraded. Let me know if you've got any questions, more than happy to help!

**Reply by fpolli** [reply]
Hi Austin, amazing work and I appreciate how active you are helping people. I am trying to choose between Canvas and Wink for my blog projects and I have a couple of questions:
1) What stack does Canvas UI use?
2) What did you finally decide re: authentication?
I am just starting to work through Laravel to build a set of related websites that will have blogs, but the blog may not necessarily be the centerpiece of the sites, and the user database needs to do more than just service the blog. I have decided to use Jetstream->Livewire. Does Canvas UI use a different stack? And would Canvas conflict with or require major modification of the underlying user system in Jetstream?
Thank you. I'm new to Laravel but I've been creating websites with PHP (and modifying WordPress sites) for a long time.

**Reply by austintoddj** [reply]
@fpolli Canvas is built similar to 1st party packages like [Horizon](https://laravel.com/docs/9.x/horizon#main-content) and [Telescope](https://laravel.com/docs/9.x/telescope#main-content).

The authentication and it's underlying tables have remained agnostic of the main application. None of Canvas' underlying structure will interfere with your existing app schema.

**Reply by fpolli** [reply]
@austintoddj If I want canvas to just use the underlying app's existing user table, how many places will I need to change code in Canvas? Is there a character sequence I can search for to bring me to all those places?

Thanks!

**Reply by austintoddj** [reply]
There's not an easy way to do that, unfortunately. Canvas user table fields are going to differ a bit from the default Laravel user table:

```php
# default
Schema::create('users', function (Blueprint $table) {
      $table->id();
      $table->string('name');
      $table->string('email')->unique();
      $table->timestamp('email_verified_at')->nullable();
      $table->string('password');
      $table->rememberToken();
      $table->timestamps();
  });

# canvas
Schema::create('canvas_users', function (Blueprint $table) {
    $table->uuid('id')->primary();
    $table->string('name');
    $table->string('email')->unique();
    $table->string('username')->unique()->nullable();
    $table->string('password');
    $table->text('summary')->nullable();
    $table->string('avatar')->nullable();
    $table->tinyInteger('dark_mode')->nullable();
    $table->tinyInteger('digest')->nullable();
    $table->string('locale')->nullable();
    $table->tinyInteger('role')->nullable();
    $table->rememberToken();
    $table->timestamps();
    $table->softDeletes();
});
```

Because of these differences and what Canvas is going to be querying against on the User model, you're not going to be able to just use the default as a drop-in replacement. 

Since this change was [introduced in 6.0.0](https://github.com/austintoddj/canvas/blob/master/.github/UPGRADE.md#upgrading-to-600-from-54), you could opt for the [5.4 release of Canvas](https://github.com/austintoddj/canvas/releases/tag/v5.4.1) which does use the default users table.

**Reply by fpolli** [reply]
Thank you, @austintoddj for the response and details. 
edit: Hi Todd, I just had a brainstorm. I can maintain the two separate user tables because my blog "users" are the contributors, and the main "users" are the subscribers. So it is entirely consistent to use two user tables. Presto Shazamo! problem solved. :)

original: In perusing release notes, there have been a lot of security fixes since 5.4 that I don't want to lose, and I don't want to lose the features of Canvas, so I'll just plow through the code and find the places for myself. Maybe I'll post a log. :) Thanks again.

**Comment by jerryhall** [comment]
Ok excellent Austin. It's uncharacteristic for me to ask so glad I did. 

Canvas looks beautiful and I can't wait to share my site in your show and tell. I hope others do the same so we can share notes. 

Thanks again!

**Reply by austintoddj** [reply]
I'm glad you reached out. If you need anything else, don't hesitate to ask.

**Comment by austintoddj** [comment]
Closing as obsolete with [Canvas v7.0.0](https://github.com/austintoddj/canvas/releases/tag/v7.0.0).
